client-rpc/src/server.rs

```diff
@@ -100,6 +102,91 @@ impl Server {
         ManualSynchronizer::new(storage, tendermint_client, block_handler)
     }
 
+    fn ask_passphrase(&self, message: Option<&str>) -> Result<SecUtf8> {
+        match message {
+            None => ask("Enter passphrase: "),
+            Some(message) => ask(message),
+        }
+        Ok(password().context(ErrorKind::IoError)?.into())
+    }
+
+    fn ask_string(&self, msg: &str, default: &str) -> String {
+        quest::ask(msg);
+        match quest::text() {
+            Ok(a) => {
+                if "" == a {
+                    default.to_string()
+                } else {
+                    a
+                }
+            }
+            Err(_b) => default.to_string(),
+        }
+    }
+
+    pub fn start_websocket(&mut self, storage: SledStorage) -> Result<()> {
+        println!("web socket");
+        let mut wallet_infos: WalletInfos = vec![];
+        let tendermint_client = RpcClient::new(&self.tendermint_url);
```

Comment:
Is this one needed if all the methods could be called over websockets?

Reply:
it's necessary to receive callbacks,
also other restful api don't have to use websockets for processing requests.

Reply:
they are not restful -- both websockets and plain HTTP requests use JSON-RPC. So it's just that there are two ways of transport.
But this could be addressed in a different PR.